lib/random.php

```diff
@@ -296,14 +273,15 @@ function random_int($min, $max)
              * Apply mask
              */
             $val &= $mask;
+            $val += $valueShift;
 
             ++$attempts;
             /**
              * If $val is larger than the maximum acceptable number for
              * $min and $max, we discard and try again.
              */
-        } while ($val > $range);
-        return (int) ($min + $val);
+        } while (!is_int($val) || $val > $max || $val < $min);
```

Comment:
The reasoning here is similar to the reasoning given when comparing strings in constant time and people complaining that the string length will leak. (See @ircmaxell's discussion with Symfony if you want to read it in detail.)

`$min` and `$max` are not sensitive values. (If they are, something is *wrong* with the design of the application.)